lib/rules/max-nesting-depth/index.js

```diff
@@ -82,9 +82,12 @@ const rule = function(max, options) {
     }
 
     if (
-      optionsMatches(options, "ignore", "blockless-at-rules") &&
-      node.type === "atrule" &&
-      node.every(child => child.type !== "decl")
+      (optionsMatches(options, "ignore", "blockless-at-rules") &&
+        node.type === "atrule" &&
+        node.every(child => child.type !== "decl")) ||
+      (optionsMatches(options, "ignore", "blockless-pseudoclasses") &&
+        node.type === "rule" &&
+        node.some(child => child.selector && child.selector.startsWith("&:")))
```

Comment:
I believe `child.selector.startsWith("&:")` will also catch [pseudo-elements](https://www.w3.org/TR/css-pseudo-4/) e.g. `::placeholder`. Can you please exclude them by checking that the 3rd character isn't a `:`?